src/python/foglamp/core/browser.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""
+Experimental browser for extracting reading data from the FogLAMP data buffer
+
+Supports a number of REST API:
+
+  http://<address>/foglamp/asset
+     - Return a summary count of all asset readings
+  http://<address>/foglamp/asset/{asset}
+    - Return a set of asset readings for the given asset 
+  http://<address>/foglamp/asset/{asset}/{reading}
+    - Return a set of sensor readings for the specified asset and sensor
+  http://<address>/foglamp/asset/{asset}/{reading}/summary
+    - Return a summary (min, max and average) for the specified asset and sensor
+
+  All but the /foglamp/asset API call take a set of optional query parameters
+    limit=x     Return the first x rows only
+    seconds=x   Limit the data return to be less than x seconds old
+    minutes=x   Limit the data returned to be less than x minutes old
+    hours=x     Limit the data returned to be less than x hours old
+
+  Note seconds, minutes and hours can not be combined in a URL. If they are then only seconds
+  will have an effect.
+
+  TODO: Improve error handling, use a connection pool
+"""
+
+import asyncpg
+import asyncio
+import json
+from aiohttp import web
+
+__author__ = "Mark Riddoch"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+__DB_URL = 'postgresql://foglamp:foglamp@localhost:5432/foglamp'
+__DEFAULT_LIMIT = 20
+__TIMESTAMP_FMT = 'YYYY-MM-DD HH24:MI:SS.MS'
+
+def setup(app):
+    """
+    Add the routes for the API endpoints supported by the data browser
+    """
+    app.router.add_route('GET', '/foglamp/asset', asset_counts)
+    app.router.add_route('GET', '/foglamp/asset/{asset_code}', asset)
+    app.router.add_route('GET', '/foglamp/asset/{asset_code}/{reading}', asset_reading)
+    app.router.add_route('GET', '/foglamp/asset/{asset_code}/{reading}/summary', asset_summary)
+
+
+async def asset_counts(request):
+    """
+    Browse all the asserts for which we have recorded readings and
+    return a readings count.
+
+    Return the result of the Postgres query 
+    select asset_code, count from readings group by asset_code;
+    """
+
+    conn = await asyncpg.connect(__DB_URL)
+
+    # Select the assets from the readings table
+    rows = await conn.fetch(
+        'select asset_code, count(*) from readings group by asset_code')
```

Comment:
`'''SELECT asset_code, COUNT(*) FROM readings GROUP BY asset_code'''`

> using triple single quote will help to avoid escaping for '{ }', we are using in next queries